### PR TITLE
Service constr exp

### PR DESCRIPTION
**Pull request by m36dot**
Add Service Constructor expression

<code>service-constructor-expr := [annots] service service-body-block</code>
<code>service-body-block := { service-method-defn* }</code>
<code>service-method-defn :=
   metadata
   [resource]
   function identifier function-signature method-defn-body</code>

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-parser/src/test/java/io/ballerinalang/compiler/parser/test/SyntaxTreeJSONGenerator.java [hunk 1]
```diff
@@ -63,7 +63,7 @@
     private static final ParserRuleContext PARSER_CONTEXT = ParserRuleContext.TOP_LEVEL_NODE;
 
     private static final PrintStream STANDARD_OUT = System.out;
-    private static final Path RESOURCE_DIRECTORY = Paths.get("src/test/resources/");
+    private static final Path RESOURCE_DIRECTORY = Paths.get("compiler/ballerina-parser/src/test/resources/");
```
Comment (SupunS) [inline compiler/ballerina-parser/src/test/java/io/ballerinalang/compiler/parser/test/SyntaxTreeJSONGenerator.java:66]: additional changes

File: compiler/ballerina-parser/src/test/resources/test1.bal [hunk 2]
```diff
@@ -1,3 +1,7 @@
-function foo() {
-    io:println("hello");
-}
+public function foo() {
+    int a =
+        service {
+            function foo() {
+            }
+        ;
+}
```
Comment (SupunS) [inline compiler/ballerina-parser/src/test/resources/test1.bal:7]: Additional changes

File: compiler/ballerina-parser/src/test/resources/expressions/xml-template/xml_template_assert_24.json [hunk 3]
```diff
@@ -306,8 +306,21 @@
...
+                                                        },
+                                                        {
+                                                          "kind": "PLUS_TOKEN",
+                                                          "isMissing": true
+                                                        },
+                                                        {
+                                                          "kind": "IDENTIFIER_TOKEN",
+                                                          "value": "b"
+                                                        }
+                                                      ]
```
Comment (SupunS) [inline compiler/ballerina-parser/src/test/resources/expressions/xml-template/xml_template_assert_24.json:323]: Why is this change required?

Comment (m36dot) [reply compiler/ballerina-parser/src/test/resources/expressions/xml-template/xml_template_assert_24.json:323]: adding this expression is causing a change in this recovery. So I changed the assert

File: compiler/ballerina-parser/src/test/java/io/ballerinalang/compiler/parser/test/SyntaxTreeJSONGenerator.java (hunk 1 above)
Comment (m36dot) [reply compiler/ballerina-parser/src/test/java/io/ballerinalang/compiler/parser/test/SyntaxTreeJSONGenerator.java:66]: will remove, didnt notice

File: compiler/ballerina-parser/src/test/resources/test1.bal (hunk 2 above)
Comment (m36dot) [reply compiler/ballerina-parser/src/test/resources/test1.bal:7]: didnt notice sorry
